autopush/websocket.py

```diff
@@ -592,12 +592,25 @@ def err_overload(self, failure, message_type):
         guarantee, so rather than never reply, we still shut the connection
         down.
 
+        If ``disconnect`` is False, the an overload error is returned and the
+        client is not disconnected.
+
+        :param disconnect: Whether the client should be disconnected or not.
+
         """
         failure.trap(ProvisionedThroughputExceededException)
-        self.transport.pauseProducing()
-        d = self.deferToLater(self.randrange(4, 9), self.err_finish_overload,
-                              message_type)
-        d.addErrback(self.trap_cancel)
+
+        if disconnect:
+            self.transport.pauseProducing()
+            d = self.deferToLater(self.randrange(4, 9),
+                                  self.err_finish_overload, message_type)
+            d.addErrback(self.trap_cancel)
+        else:
+            send = {"messageType": "error",
+                    "reason": "overloaded",
+                    "status": 503
+                    }
```

Comment:
How does the client know when to allow requests again—should it just use exponential backoff?